xr_sample_project/Assets/SampleViewer/Samples/VRSample/Scripts/SetWeatherType.cs

```diff
@@ -9,51 +9,43 @@
 
 public class SetWeatherType : MonoBehaviour
 {
-    private Volume volume;
-    private VolumetricClouds volumetricClouds; 
-    private GameObject weatherParticles;
-
+    [SerializeField] WeatherSystem weatherSystem;
     private void Start()
     {
-        weatherParticles = GameObject.FindWithTag("WeatherParticles");
-
         // Delay on volume cache to give it time to instantiate
-        Invoke(nameof(GetVolume), 0.5f);
+        Invoke(nameof(InitializeWeather), 0.5f);
     }
 
-    private void GetVolume()
+    private void InitializeWeather()
     {
-        volume = FindObjectOfType<Volume>();
-        if (volume)
+        weatherSystem = FindAnyObjectByType<WeatherSystem>();
+        if(weatherSystem)
         {
-            // Start with sunny weather
             SetWeatherTypeFromIndex(0);
         }
     }
 
     public void SetWeatherTypeFromIndex(int index)
     {
-
-        weatherParticles = weatherParticles ? weatherParticles : GameObject.FindWithTag("WeatherParticles");
-        volume = volume ? volume : FindObjectOfType<Volume>();
-
         if (index == 0) //Sunny
         {
-            weatherParticles.SetActive(false);
-            if (volume.profile.TryGet<VolumetricClouds>(out volumetricClouds))
-            {
-                volumetricClouds.enable.overrideState = true;
-                volumetricClouds.enable.value = false;
-            }
+            weatherSystem.SetToSunny();
```

Comment:
```suggestion
            weatherSystem?.SetToSunny();
```
If you change these calls, you wont need to check for the weather system on line 22. Will be safer anyway.